src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -326,12 +326,8 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
         Activity? activity = null;
         if (_activitySource.HasListeners())
         {
-            if (ActivityContext.TryParse(requestId, traceState, true, out ActivityContext context))
+            if (ActivityContext.TryParse(requestId, traceState, isRemote: true, out ActivityContext context))
             {
```

Comment:
```suggestion
            if (ActivityContext.TryParse(requestId, traceState, isRemote: true, out ActivityContext context))
            {
                // The requestId used the W3C ID format, so we use the ActivityContext created from the requestId
                // as the parent context to make it the parent activity.
```

Is this better @adityamandaleeka?

Reply:
I think we should keep the part about IsRemote since that's the main reason why we use the context-based approach (https://github.com/dotnet/aspnetcore/pull/41568#discussion_r868733305)

I'd say leave it as-is, unless you want to clarify how TryParse uses the isRemote (but IMO that's too much detail).

Reply:
The "and setting ActivityContext.IsRemote to true after parsing it" isn't happening anymore, so I think we should at least remove that part.

"ActivitySource.CreateActivity overload that takes a string ID sets ActivityContext.IsRemote = false when it parses the string internally" is also a little bit confusing considering that it doesn't look like `ActivitySource.CreateActivity` creates an `ActivityContext`. I assume it's referring to this logic that sets the publicly gettable `HasRemoteParent` property.

https://github.com/dotnet/runtime/blob/e4ac5cd051107772b99cf3f275f2718ff20120a4/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L1117-L1133

How about this then?

```suggestion
            if (ActivityContext.TryParse(requestId, traceState, isRemote: true, out ActivityContext context))
            {
                // The requestId used the W3C ID format. Unfortunately, the ActivitySource.CreateActivity overload that
                // takes a string parentId never sets HasRemoteParent to true. We work around that by calling the
                // ActivityContext overload instead which sets HasRemoteParent to parentContext.IsRemote.
                // https://github.com/dotnet/aspnetcore/pull/41568#discussion_r868733305
```

Reply:
Sure, that works.